docs/source/specs/typespec/main.tsp

```diff
@@ -103,21 +103,33 @@ namespace Workspaces {
         ...Timestamps;
     }
 
+    model WorkspaceAncestry {
+        @key uuid: UUID;
+        parent_id?: UUID;
+        name: string;
+    }
+
     alias CreateWorkspaceRequest = BasicWorkspace;
 
-    model CreateWorkspaceResponse extends BasicWorkspace{
+    model CreateWorkspaceResponse extends Workspace{
         @statusCode _: 201;
     }
 
-    model ReadWorkspaceResponse extends BasicWorkspace{
+    model ReadWorkspaceResponse extends Workspace {
         @statusCode _: 200;
     }
 
+    model ReadWorkspaceWithAncestryResponse extends ReadWorkspaceResponse {
+        ancestry: WorkspaceAncestry[];
+    }
+
     model UpdateWorkspaceRequest {
         @doc("Workspace A")
         name: string = "Workspace A";
         @doc("Description of Workspace A")
-        description?: string = "Description of Workspace A";
+        description: string = "Description of Workspace A";
+        @doc("Description of Workspace A")
+        parent_id: UUID = "Parent UUID of Workspace A";
```

Comment:
@alechenninger that all makes total sense. This specifically was to fix discrepancies between the current API behavior and the API spec so that clients can actually be built (currently `parent_id` is required in the update). I totally agree with what you're saying, and if it's good with you I'd prefer to log a ticket to _prevent_ `parent_id` from being changed, and not require it in PUT/PATCH.

Reply:
Logged a shell ticket here: https://issues.redhat.com/browse/RHCLOUD-35415